Making antag objective rolls not suck, adding 2 new objectives, making revs start with 2 headrevs minimum.

RorororoIDK opened the pull request:
Tweaking objective weights and probability so traitors and thieves get more engaging objectives as well as more overall, adding 2 new rp objectives and rewording others for more wiggle room, as well as adding elite suit and eshield to uplink and making revs have a minimum of 2 headrevs.


## Short description
Changing objective probabilities for traitor so they get better ones more often, lowering the difficulty for objectives on both so they can roll more objectives, adding elite suit to uplink because like, why not, its not overpowered, same with eshield, added 2 new rp objectves and lowered the difficulty of most rp objectives, also making revs have a minimum of 2 headrevs because one headrev on like 90 pop is ass and boring

## Why we need to add this
Finally rolling traitor just to get "Help X complete objective" or "Steal prime cut corgi meat" is never fun, these 2 for example had objective weights so high it took up over half your max difficulty which is insane, License and Diploma steal objectives also did this, for context, they was almost as high as killing a command member for an objective. 

Lowered the objective weights for thieves since thief is really not that hard and this lets them do more stuff. 

Added the elite suit to the uplink was kinda just, a side thing because there is like, no reason traitors shouldn't have it, its a sidegrade and quite expensive, also added the energy shield because frankly its not that insane and it immediately makes you kos, it breaks under kinetic fire quite easily.

Buffed the Elite suit and lowered its price slightly as it was brought up in a suggestion that its 12tc for an objectively worse hardsuit, theres no reason this should cost the same as the jug, buffed its resists to 40% blunt, 40% slash and 90% heat, lasers do 2 damage to you now instead of 4, it now costs 10tc.

Gave ERT security hardsuit its own stats, slightly weaker than a default bloodred for the reason that ert gets better weaponry and is usually arriving to fight a weakened force, ERT suit is no longer a direct copy of Elite bloodred.

Lowered the difficulty for most RP objectives as they were rather high for what most of them were, as well as rewording a few so you get do some more chaotic shit with them. 

Made revs start with a minimum of 2 headrevs, because rolling with a singular headrev was ass, most of the time on revs theres one headrev who never gets converts and frankly the responsibility of an entire major antag gamemode relying solely on one person is not realistic, Im just praying this change works as no other gamerule has a minimum antag amount, if broken, its one line of code. 

Lastly, I upped the chance to roll DAGD, why? because an under 0.9% chance is insanely low and most of the people who get this objective never use it to its full potential, or at all, let traitors be chaotic, we have rules for no loosing so let it play out and see how it goes.

## Media (Video/Screenshots)
<!--
If your PR contains in-game changes you must provide screenshots/videos of the changes.
-->

## Checks
<!-- check boxes for faster reviewing of your PR -->

- [x] I do not require assistance to complete the PR.
- [x] Before posting/requesting review of a PR, I have verified that the changes work.
- [ ] I have added screenshots/videos of the changes, or this PR does not change in-game mechanics.
- [x] I affirm that my changes are licensed under the [Starlight Fork License](https://github.com/ss14Starlight/space-station-14/blob/Starlight/LICENSE-Starlight.TXT) and grant permission for use in this repository under its conditions.

**Changelog**

:cl: Roro
- add: 2 new RP objectives that let you be an absolute nuisance
- tweak: Lowered difficulty for most traitor and thief objectives so they can roll more objectives together, also tweaked some to be higher but only where it made actual sense
- tweak: Tweaked the probability of rolling some objectives, too many to list here, tldr, you roll more engaging ones and less "Help X with objective"
- tweak: You can no longer roll multiple help objectives, its boring, you get one, that's it.
- tweak: Added elite suit and energy shield to uplink
- tweak: Buffed the elite suits resists slightly and lowered its cost (no reason it should have costed same as jugsuit), buffed to 40% blunt, 40% slash and 90% heat, now costs 10tc
- tweak: Made revs start with a minimum of 2 headrevs (probably works, I've been told it should)
- tweak: Some RP objectives have been reworded so you can do more with them 
- tweak: Die a glorious death is now much more common but also limited to a max of 3 players per round. Up from an under 0.9% chance to around a 3.5%-ish chance
- tweak: ERT security hardsuit now has its own stats rather than being a complete copy of Elite bloodred (ERT sec can now die to explosives), now more in line with Default bloodred but slightly weaker since ERT get better weapons

RorororoIDK committed:
Tweaking objective chances and weights

Tweaking objective weights and probability so traitors and thieves get more engaging objectives as well as more overall, adding 2 new rp objectives and rewording others for more wiggle room, as well as adding elite suit to uplink and making revs have a minimum of 2 headrevs.

RorororoIDK committed:
Add eshield to uplink

I want chaos, also its not that broken, it counters lasers but breaks under kinetic pretty easily

maxion12345 commented:
Not sure I agree with giving nuclear operative gear straight in the uplink. These are usually a surplus crate only item, since you aren't meant to be an infiltrator and not "budget nukeop". I guess e-shield sort of makes sense since it's concealable, but why would the Syndicate provide you with an highly armored hardsuit if you're meant to be stealing and assassinating with all the covert tools they provide?

Happyrobot33 committed:
Merge branch 'starlight-dev' into Antag-weights

RorororoIDK commented:
> Not sure I agree with giving nuclear operative gear straight in the uplink. These are usually a surplus crate only item, since you aren't meant to be an infiltrator and not "budget nukeop". I guess e-shield sort of makes sense since it's concealable, but why would the Syndicate provide you with an highly armored hardsuit if you're meant to be stealing and assassinating with all the covert tools they provide?

It costs half their tc and is a sidegrade, the elite suit is not highly armored, its good against lasers and thats it, it has a 10% slow and lower pierce, blunt and slash resist than a default bloodred and infiltrator suit, also its not a "budget nukeop" a budget nukeop would be me giving them raid suit or jug suit which im not, those are exclusively nukeop hardsuits, the elite suits has the worst general armor out of all the hardsuits

RorororoIDK committed:
Merge branch 'starlight-dev' into Antag-weights

Arkanic commented:
wizden has/had elite suits purchaseable, it's not unheard of

might be worth checking why it got removed (ie was it a "this is so op wtf" back when it didn't have like some pitifully low pierce resist)

either way i don't really see the problem given that you can buy full bloodreds anyway, elite is more of a sidegrade than an upgrade to bloodred

RorororoIDK commented:
> > Not sure I agree with giving nuclear operative gear straight in the uplink. These are usually a surplus crate only item, since you aren't meant to be an infiltrator and not "budget nukeop". I guess e-shield sort of makes sense since it's concealable, but why would the Syndicate provide you with an highly armored hardsuit if you're meant to be stealing and assassinating with all the covert tools they provide?
> 
> It costs half their tc and is a sidegrade, the elite suit is not highly armored, its good against lasers and thats it, it has a 10% slow and lower pierce, blunt and slash resist than a default bloodred and infiltrator suit, also its not a "budget nukeop" a budget nukeop would be me giving them raid suit or jug suit which im not, those are exclusively nukeop hardsuits, the elite suits has the worst general armor out of all the hardsuits

Default bloodred is 10% slow, 50% blunt, 50% slash, 50%, pierce, 50% heat Infiltrator is NO slow, 40% blunt, 40% slash, 40% pierce, 40% heat and over 2 minutes of invisibility and finally Elite suit is the worst with 10% slow, 30% blunt, 30% slash, 30% pierce, but 80% heat, this thing costs the most and is only good against one type of damage for half of your tc, its arguably a downgrade and very situational

RorororoIDK committed:
Making elite actually viable

Upping its blunt slash and heat a little so its more anti laser and doesnt get eviscerated by melee

RorororoIDK committed:
Lowering its price slightly

lowering elites price so its not the same as a jug suit..

RorororoIDK committed:
Updating ERT security hardsuit stats

ERT security hardsuit was a copy of the elite bloodred hardsuit, im changing this so the suit has its own stats and is more in line with a default bloodred

RorororoIDK committed:
Merge branch 'starlight-dev' into Antag-weights

RorororoIDK committed:
it duped fml, fixed

RorororoIDK committed:
No more multiple help objectives

AndrewTateOfficialSS14 commented:
it is so relatable about DAGD, i usually see people roll it and be ALIVE at the end of the round, like wtf, the most interesting objective and yet they do 0 shit

AndrewTateOfficialSS14 commented:
it seems like DAGD started rolling way LESS than it used to, 3 traitor rounds in a row noone rolled it, i highly doubt that limiting it to 3 people will do anything, considering how unlikely it is to be rolled. Maybe mind upping its chances but restricting to 1 person only?
